test/integration/limited_ciphers_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"crypto/tls"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"go.pinniped.dev/internal/config/supervisor"
+	"go.pinniped.dev/test/testlib"
+)
+
+// TestLimitedCiphers_Disruptive will confirm that the Pinniped Supervisor exposes only those ciphers listed in
+// configuration.
+// This does not test the Concierge (which has the same feature) since the Concierge does not have exposed API
```

Comment:
Couldn't this also test the Supervisor's aggregated API port and the Concierge's aggregated API port?

Reply:
It's my understanding that those ports will always use `ptls.Secure` which is not influenced by this new configuration option. I'll go back and check.

Reply:
Oh I see what you mean. In non-FIPS mode they will use Secure which only allows TLS 1.3. On the other hand, in FIPS mode they will still use Secure, but in FIPS mode Secure will use TLS 1.2 and be influenced by this new configuration option. This test runs in both FIPS and non-FIPS mode, so it would be possible to test those other ports if the assertions were dynamic based on FIPS vs. non-FIPS.

Reply:
What did you think about this?

Reply:
Added in this branch.